# Data Science Tools and Ecosystem

In this notebook, we will explore and summarize the various tools and ecosystem available to data scientists, including popular programming languages, software libraries, frameworks, and data storage options. Whether you are just starting out in data science or looking to enhance your skills, this notebook aims to serve as a helpful resource in navigating the vast landscape of data science tools and technologies.

Some of the popular languages that Data Scientists use are:

1. Python
2. R
3. SQL

Python is a versatile language frequently used in data cleaning, exploration, pre-processing and modeling. R is another popular language that also focuses on data exploration and data visualization. SQL is a language that is used to manage, manipulate and query databases. These three are the most commonly used programming languages in Data Science.

Some of the commonly used libraries used by Data Scientists include:

- Pandas
- Scikit-Learn
- NumPy

Pandas is a library used for data manipulation and analysis. With Pandas, you can easily import and export data from different formats and sources such as CSV, TSV, Excel, SQL databases, and more. It provides datastructures such as dataframes and series.

Scikit-Learn is a library used for machine learning in Python. It provides a variety of simple and efficient machine learning algorithms and tools for data mining and data analysis.

NumPy is a library used for scientific computing in Python. It provides N-dimensional array objects which are more efficient than Python's built-in lists and arrays for performing mathematical operations on large datasets.

# Data Science Tools

- Jupyter Notebook
- RStudio
- Spyder

In the table above, the header "Data Science Tools" is in the first row and is centered. The subsequent three rows list open source tools used in data science: Jupyter Notebook, RStudio, and Spyder.

# Examples of Evaluating Arithmetic Expressions in Python

In Python, arithmetic expressions can be evaluated using mathematical operators such as addition (+), subtraction (-), multiplication (*), division (/), and modulo (%). Below are some examples of evaluating arithmetic expressions in Python:

- Addition:

In [1]:
x = 5
y = 3
result = x + y
print(result)

8


- Subtraction:

In [2]:
a = 10
b = 7
result = a - b
print(result)

3


- Multiplication:

In [3]:
m = 3
n = 5
result = m * n
print(result)

15


- Division:

In [4]:
p = 20
q = 5
result = p / q
print(result)

4.0


Modulo:

In [5]:
r = 10
s = 3
result = r % s
print(result)

1


In [6]:
# This is a simple arithmetic expression to multiply 3 and 4, then add 5
(3 * 4) + 5

17

In [7]:
# This will convert 200 minutes to hours by dividing by 60.
hours = 200 / 60
print(hours)

3.3333333333333335


When you run this cell, it will output the result of the expression, which is 3.3333333333333335. This means that 200 minutes is equivalent to 3.33 hours (rounded to two decimal places).

# Objectives:

- List popular languages for Data Science
- Understand the Data Science workflow: Data collection, Data cleaning, Data exploration, Data visualization, and Data modelling
- Perform basic data manipulation using Pandas library
- Use Seaborn library for data visualization
- Understand the basic concepts of Machine Learning and its types
- Train and evaluate a Classification model using Scikit-Learn library

# Author

Reagan Mayoko, as Data Scientist